### PR TITLE
docs: Use @code/@endcode.

### PR DESCRIPTION
**Pull request by waywardmonkeys**
Use `@code` and `@endcode` markers in the doxygen comments.

Also, use an empty line before code blocks to help ensure they're picked up correctly. When using `@code`...`@endcode` markers, don't indent the code.

### CODE REVIEW

File: include/flecs.h [hunk 1]
```diff
@@ -2971,14 +2986,16 @@ bool ecs_is_valid(
...
  *
- *   ecs_entity_t e1 = ecs_new_id(world);
- *   ecs_is_alive(world, e1);             // true
- *   ecs_delete(world, e1);
- *   ecs_is_alive(world, e1);             // false
+ * @code
+ * ecs_entity_t e1 = ecs_new_id(world);
+ * ecs_is_alive(world, e1);             // true
+ * ecs_delete(world, e1);
+ * ecs_is_alive(world, e1);             // false
```
Comment (SanderMertens) [inline include/flecs.h:2993]: Did you forget an endcode here?

File: include/flecs/addons/cpp/entity_view.hpp [hunk 2]
```diff
@@ -354,13 +375,16 @@ struct entity_view : public id {
...
      * 
      * Example:
-     *   e.get([](Position& p, Velocity& v) { // write lock
-     *     p.x += v.x;
-     *   });
+     *
+     * @code
+     * e.get([](Position& p, Velocity& v) { // write lock
+     *   p.x += v.x;
+     * });
```
Comment (SanderMertens) [inline include/flecs/addons/cpp/entity_view.hpp:382]: Did you forget an endcode here?

**Review by SanderMertens — CHANGES_REQUESTED** [review]

**Comment by waywardmonkeys** [comment]
@SanderMertens In both cases, the `@endcode` was a couple of lines down due to a blank line in the snippet. No changes should be needed to fix that.

**Review by SanderMertens — APPROVED** [review]

**Comment by SanderMertens** [comment]
Awesome, thanks for the PR!

**Pull request merged**